lib: lte_link_control: Remove CONFIG_LTE_NETWORK_USE_FALLBACK (#16635)
nrfconnect/sdk-nrf: doc/nrf/libraries/modem/lte_lc.rst

For additional configurations related to these features, see the API documentation.

Connection fallback
===================
It is possible to try to switch between LTE-M and NB-IoT if a connection has not been established.
This is useful when the connection to either of these networks becomes unavailable.

> Use the :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT` or :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT_GPS` Kconfig option
> with which the modem will perform a fallback from one system mode to another if a connection to the first one fails.

Reviewer: ```suggestion
Use the :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT` or :kconfig:option:`CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT_GPS` Kconfig option to make the modem perform a fallback from one system mode to another if a connection to the first one fails.
```

Author: This is now removed after discussing with my colleagues

In addition, you can control the priority between LTE-M and NB-IoT using the :kconfig:option:`CONFIG_LTE_MODE_PREFERENCE` Kconfig option.

> You can also configure the the overall connection establishment time with the :kconfig:option:`CONFIG_LTE_NETWORK_TIMEOUT` Kconfig option.

Reviewer: ```suggestion
You can also configure the overall connection establishment time with the :kconfig:option:`CONFIG_LTE_NETWORK_TIMEOUT` Kconfig option.
```

Author: This is now removed after discussing with my colleagues
